core/ecschema-metadata/src/Metadata/Schema.ts

```diff
@@ -443,22 +443,225 @@ export class Schema implements CustomAttributeContainerProps {
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested PropertyCategory or undefined if not found.
+   */
+  public async getPropertyCategory(name: string): Promise<PropertyCategory | undefined> { return this.getItem(name, PropertyCategory); }
+
+  /**
+   * Shortcut for calling getItem with Unit.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested Unit or undefined if not found.
+   */
+  public async getUnit(name: string): Promise<Unit | undefined> { return this.getItem(name, Unit); }
+
+  /**
+   * Shortcut for calling getItem with InvertedUnit.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested InvertedUnit or undefined if not found.
+   */
+  public async getInvertedUnit(name: string): Promise<InvertedUnit | undefined> { return this.getItem(name, InvertedUnit); }
+
+  /**
+   * Shortcut for calling getItem with Constant.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested Constant or undefined if not found.
+   */
+  public async getConstant(name: string): Promise<Constant | undefined> { return this.getItem(name, Constant); }
+
+  /**
+   * Shortcut for calling getItem with Phenomenon.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested Phenomenon or undefined if not found.
+   */
+  public async getPhenomenon(name: string): Promise<Phenomenon | undefined> { return this.getItem(name, Phenomenon); }
+
+  /**
+   * Shortcut for calling getItem with UnitSystem.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested UnitSystem or undefined if not found.
+   */
+  public async getUnitSystem(name: string): Promise<UnitSystem | undefined> { return this.getItem(name, UnitSystem); }
+
+  /**
+   * Shortcut for calling getItem with Format.
+   * @param name The local (unqualified) name of the item to return.
+   * @returns The requested Format or undefined if not found.
+   */
+  public async getFormat(name: string): Promise<Format | undefined> { return this.getItem(name, Format); }
   /**
    * Gets an item from within this schema. To get by full name use lookupItem instead.
    * @param key the local (unqualified) name, lookup is case-insensitive
+   * @param itemConstructor optional, the constructor of the item to return. If missing, the returned value cannot be type checked.
    */
-  public async getItem<T extends SchemaItem>(name: string): Promise<T | undefined> {
+  public async getItem<T extends SchemaItem>(name: string, itemConstructor?: SchemaItemConstructor<T>): Promise<T | undefined> {
     // this method exists so we can rewire it later when we load partial schemas, for now it is identical to the sync version
-    return this.getItemSync<T>(name);
+    return this.getItemSync<T>(name, itemConstructor);
   }
 
+  /**
```

Comment:
can we avoid adding these sync methods?  If we want to prioritize the async methods then the getItemSync method should be enough for callers

Reply:
Yeah makes sense. Just seemed a bit strange since getting items within a schema is always sync at the moment and I'm not sure how likely we are to implement partially loaded schemas with the current class design.... So we could also do away with the async ones and leave that to ISchemaItemLocater. What do you think?

Reply:
Hm, could we also consider the opposite? The items returned by these methods are all from within this schema, and the schema is always fully loaded, so accessing its items will always be sync, no?
If at some point we implement a partially loaded schema, we will probably do that in a separate class, not reusing the existing Schema class.
We do have ISchemaItemProvider which could serve that purpose.

Reply:
done